awx/main/management/commands/check_instance_ready.py

```diff
@@ -0,0 +1,12 @@
+from django.core.management.base import BaseCommand
+from awx.main.models.ha import Instance
+
+
+class Command(BaseCommand):
+    help = 'Check if the task manager instance is ready throw error if not ready, can be use as readiness probe for k8s.'
+
+    def handle(self, *args, **options):
+        if Instance.objects.me().node_state != Instance.States.READY:
+            raise Exception('Instance is not ready')  # so that return code is not 0
```

Comment:
The correct way to exit out of Django commands with a failure is to raise `CommandError` (from `django.core.management.base`)